src/cmd/cloud.js

```diff
@@ -378,16 +412,31 @@ module.exports = class CloudCommand extends CLICommandBase {
 					throw normalizedApiError(resp);
 				}
 			})
-			.catch(err => {
-				throw normalizedApiError(err);
-			})
 			.then((sizeInfo) => {
-				if (sizeInfo){
+				respSizeInfo = sizeInfo;
+				if (assets) {
+					return new BundleCommands()._generateBundle({ assetsList: assets, appBinary: filename, bundleFilename: bundleFilename });
+				}
+			})
+			.then((bundle) => {
+				if (respSizeInfo){
 					this.ui.stdout.write(`Memory use:${os.EOL}`);
-					this.ui.stdout.write(`${sizeInfo}${os.EOL}`);
+					this.ui.stdout.write(`${respSizeInfo}${os.EOL}`);
+				}
+				if (bundle) {
+					if (fs.existsSync(filename)){
+						fs.unlinkSync(filename);
+					}
+					// TODO: Verify that the bundle is created with the correct files
```

Comment:
Remember to remove this comment before merging the PR

Reply:
I was thinking along of lines of using BVR's `unpackApplicationAndAssetBundle` over here but wanted to ask for your suggestion before doing that. If it's not needed to be too specific, I will remove it.